src/main/java/xyz/nucleoid/plasmid/game/GameOpenContext.java

```diff
@@ -14,8 +14,8 @@
         this.game = game;
     }
 
-    public CompletableFuture<GameWorld> openWorld(BubbleWorldConfig config) {
-        return GameWorld.open(this.server, this.game, config);
+    public GameOpenProcedure opens(BubbleWorldConfig worldConfig, Consumer<GameLogic> configureGame) {
```

Comment:
```suggestion
    public GameOpenProcedure open(BubbleWorldConfig worldConfig, Consumer<GameLogic> configureGame) {
```

Not sure why there's a `s` here.

Reply:
It isn't doing the actual opening, but declaring that "this opens like this"

Reply:
Then maybe renaming this method `onOpen`, `whenOpening` maybe? I'm not sure.

Reply:
Verbose option would be to go with `createOpenProcedure`?

Reply:
Could work